File: 717.c
bool isOneBitCharacter(int* bits, int bitsSize) {
    int possible[bitsSize];
    memset(possible, 0, sizeof(int) * bitsSize);
    possible[0] = 1;
    for (int i = 0; i < bitsSize - 1; i++) {
        if (possible[i]) {
            if (bits[i] == 0) {
                possible[i + 1] = 1;
            } else if (i + 2 < bitsSize) {
                possible[i + 2] = 1;
            }
        }
    }
    return possible[bitsSize-1];
}

File: 717.py
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        even = True
        for i in range(len(bits) - 1):
            if even and bits[i] == 0 and i % 2 == 0:
                even = not even
            elif not even and bits[i] == 0 and i % 2 == 1:
                even = not even
        if len(bits) % 2 == 0:
            return not even
        else:
            return even
